### PR TITLE
[API-8] Expose PaginationCalculator

### LINKED ISSUE #2138: Expose Pagination API/PaginationCalculator
It would be nice if the `PaginationCalculator` was exposed by the API, so we could use it to make centered texts, add padding to texts and etc. Since the class is private in SpongeCommon, we can't even add it to the classpath to be able to use the class, the only way being copying the whole class.

### PR DESCRIPTION
**Pull request by ItsDoot**
Closes #2138.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/service/pagination/PaginationCalculator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package org.spongepowered.api.service.pagination;
```
Comment (Zidane) [inline src/main/java/org/spongepowered/api/service/pagination/PaginationCalculator.java:1]: Forgot license headers.

**Comment by Zidane** [comment]
Also, this could target api-7 as well.

**Comment by simon816** [comment]
I think it's worth noting in the documentation that this assumes the vanilla font. In ChatUILib I have [`FontData`](https://github.com/simon816/ChatUI/blob/ba4fc94457d6dc741e6b7bafe0605096f7a6e5af/ChatUILib/src/main/java/com/simon816/chatui/util/FontData.java) which reads a .png to determine character sizes, allowing non-vanilla ascii.png measurements.
For unicode characters there's `glyph_sizes.bin`, though I'm not sure if this is modifiable in a datapack.

**Comment by simon816** [comment]
Also, to centre text means knowing the width of the chat box. SpongeCommon [assumes the default](https://github.com/SpongePowered/SpongeCommon/blob/7ea7c5d8e69f84fa9b1c1c3eddb90d4fff5d4dd7/src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java#L56) (320), but it is customizable (see https://ore.spongepowered.org/simon816/Chat-UI-Library#player-settings)

**Comment by zml2008** [comment]
I think it's entirely fine that the Sponge Common implementation only supports vanilla content. Perhaps it could be expanded to be able to load data from resource packs and automatically configure based on the server pack, but it *is* a service and can be replaced by other more customizable implementations.

Another theoretical implementation could integrate with a client mod that passed the necessary sizing and resource pack information to the server,but that would be imo out of scope for sponge itself. However, to make this more friendly to such an implementation, the `getCalculator` method should probably take a `Player` parameter?

**Comment by dualspiral** [comment]
This has sat around for some time - what do we want to do with this?

**Comment by ImMorpheus** [comment]
Hey! You seem to be running an unsupported version (API 8) of SpongeAPI. Support for API 8 ended in October 2023.

API-8 is now end-of-life. We will not be issuing any more API updates for version 8 of our API[^1]

As part of our legacy cleanup we are closing issues relating to API8/1.16
___

If you wish to move this forward, please rebase this PR on the current branch and reopen it.

[^1]: https://forums.spongepowered.org/t/sponge-status-update-17th-october-2023/42212#spongeapi-82-1

**Pull request closed without merging**